singer_sdk/target_base.py

```diff
@@ -332,6 +332,7 @@ def _process_schema_message(self, message_dict: dict) -> None:
             message_dict: The newly received schema message.
         """
         self._assert_line_requires(message_dict, requires={"stream", "schema"})
+        self._assert_line_requires(message_dict["schema"], requires={"properties"})
```

Comment:
The exception raised here (`Exception`) is too generic. A more concrete exception class would be nice